v2/tests/database_transactionsjs_test.go

```diff
@@ -0,0 +1,76 @@
+//
+// DISCLAIMER
+//
+// Copyright 2023 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+
+package tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/arangodb/go-driver/v2/arangodb"
+)
+
+func Test_DatabaseTransactionsJS(t *testing.T) {
+	Wrap(t, func(t *testing.T, client arangodb.Client) {
+		WithDatabase(t, client, nil, func(db arangodb.Database) {
+
+			t.Run("Transaction ReturnValue", func(t *testing.T) {
+				WithCollection(t, db, nil, func(col arangodb.Collection) {
+					withContextT(t, 30*time.Second, func(ctx context.Context, t testing.TB) {
+						txJSOptions := arangodb.TransactionJSOptions{
+							Action: "function () { return 'worked!'; }",
+							Collections: arangodb.TransactionCollections{
+								Read:      []string{col.Name()},
+								Write:     []string{col.Name()},
+								Exclusive: []string{col.Name()},
+							},
+						}
+
+						result, err := db.TransactionJS(ctx, txJSOptions)
+						require.NoError(t, err)
+						require.Equal(t, "worked!", result)
+					})
+				})
+			})
+
+			t.Run("Transaction ReturnError", func(t *testing.T) {
```

Comment:
a good test-case here might be fetching command line options which were used during arangod execution. This is used in arangosync (driver v1):
```

	jsAction := `
function() {
    return require("internal").options();
}`
	actionResult, err := db.Transaction(ctx, jsAction, nil)
	if err != nil {
		return defaults, errors.WithMessage(err, "Transaction failed")
	}

	optionsMap, ok := actionResult.(map[string]interface{})
	if !ok {
		return defaults, fmt.Errorf("could not parse transaction result: %+v", actionResult)
	}

	minRF, _ := utils.GetFieldFromMap(optionsMap, "cluster.min-replication-factor", float64(defaults.ClusterMinReplicationFactor))
	maxRF, _ := utils.GetFieldFromMap(optionsMap, "cluster.max-replication-factor", float64(defaults.ClusterMaxReplicationFactor))
	forceOneShard, _ := utils.GetFieldFromMap(optionsMap, "cluster.force-one-shard", defaults.ForceOneShard)

```

Reply:
added